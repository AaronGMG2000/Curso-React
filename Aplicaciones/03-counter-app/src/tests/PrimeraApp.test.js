import React from "react";
import { shallow } from "enzyme"
import PrimeraApp from '../PrimeraApp';

describe('pruebas en <PrimeraApp/>', () => {
    test('debe mostrar <PrimeraApp/> correctamente', () => {
        const saludo = "hola";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = "hola";
        const subtitulo = "Esto es un subtitulo"
        const wrapper = shallow(<PrimeraApp 
            saludo={saludo}
            subtitulo = {subtitulo}
        />);
        
        const textParrafo = wrapper.find('p').text();
        expect(textParrafo).toBe(subtitulo);
    });
})
