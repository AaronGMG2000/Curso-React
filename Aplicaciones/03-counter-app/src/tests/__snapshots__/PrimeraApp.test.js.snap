// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <PrimeraApp/> debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    hola
     
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;

exports[`pruebas en <PrimeraApp/> debe mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
     
    hola
     
  </h1>
  <p>
    Esto es un subtitulo
  </p>
</Fragment>
`;
